t/unit/transport/test_redis.py

```diff
@@ -695,6 +695,10 @@ def test_connparams_regular_hostname(self):
         self.channel.connection.client.hostname = 'george.vandelay.com'
         assert self.channel._connparams()['host'] == 'george.vandelay.com'
 
+    def test_connparams_username(self):
+        self.channel.connection.client.userid = 'kombu'
+        assert self.channel._connparams()['username'] == 'kombu'
+
     def test_connparams_password_for_unix_socket(self):
         self.channel.connection.client.hostname = \
             'socket://:foo@/var/run/redis.sock'
```

Comment:
Could you also add a test here to `test_connparams_password_for_unix_socket` that tests the parsing of the username from the hostname string?

Reply:
@pomegranited I went with a different approach but still testing to get the username and password from the connection url.